projects/plugins/backup/src/php/class-jetpack-backup.php

```diff
@@ -132,4 +134,106 @@ private function get_initial_state() {
 		require_once JETPACK_BACKUP_PLUGIN_DIR . '/src//php/class-initial-state.php';
 		return ( new Initial_State() )->render();
 	}
+
+	/**
+	 * Register REST API
+	 */
+	public function register_rest_routes() {
+
+		// Get information on most recent 10 backups.
+		register_rest_route(
+			'jetpack/v4',
+			'/backups',
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => __CLASS__ . '::get_recent_backups',
+				'permission_callback' => __CLASS__ . '::backups_permissions_callback',
+			)
+		);
+
+		// Get site backup/scan/anti-spam capabilities.
+		register_rest_route(
+			'jetpack/v4',
+			'/backup-capabilities',
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => __CLASS__ . '::get_backup_capabilities',
+				'permission_callback' => __CLASS__ . '::backups_permissions_callback',
+			)
+		);
+
+	}
+
+	/**
+	 * The backup calls should only occur from a signed in admin user
+	 *
+	 * @access public
+	 * @static
+	 *
+	 * @return true|WP_Error
+	 */
+	public static function backups_permissions_callback() {
```

Comment:
@fgiannar Do you think this is necessary, or just allow the public-api to handle the auth?  If it's reasonable to put something here, do you have thoughts on what we should check?  We may also want to check auth on the two end points I'm using, one is site, the other is user.  If we review and want to change them later we'd break the `request_as_blog` or `request_as_user` calls.

Reply:
Hey John!

> Do you think this is necessary, or just allow the public-api to handle the auth? If it's reasonable to put something here, do you have thoughts on what we should check? 

I think having permission checks here, based on the current user's capabilities makes sense! We could check for example for the `manage_options` capability.

> We may also want to check auth on the two end points I'm using, one is site, the other is user. If we review and want to change them later we'd break the request_as_blog or request_as_user calls.

Since in our context here, both endpoints will be used for actions on behalf of the current user, I'd suggest using `request_as_user` in both cases.

Reply:
Ok, we can then leave it as is for now and address this in a future PR when we have complete Sync functionality on WPCOM.